text/0000-canonical-metadata.md

```diff
@@ -1216,6 +1216,146 @@ interface StaticResourceReference {
 ```
 </details>
 
+## Metadata Shape Option 2
+
+The main differences in proposed shape #2:
+- the root object interface:  not only the root object contains metadata for each file, it also describes the shape of an overall bundle/component via interface object, which contains public properties, slots, events or anything that is accessible from the outside of the component/module. The reason for doing so is that individual file metadata doesn't have the notion of other files and therefore cannot make assumption of whether it is a component. 
+- no decorators
+- file metadata revolves around declarations (both internal and external). The reason for this is that we don't know the content of the file, thus the metadata is collected without any assumptions - could be module with few _const_ exports, could be a class, etc.
+
+### TypeScript
+
+#### Root Metadata Object
+```ts
+interface BundleMetadata {
+    success: boolean;
+    version: string;
+    diagnostics: Array<Diagnostic>;
+    entry: string,
+    result: { [name: string]: FileMetadata } // metadata per each source
+    [api or interface]: ComponentInterface, // public methods, events, css properties, exports, slots.
```

Comment:
Document that this is a simpler way for an external user to see the public API snapshot of a given bundle without having to drill down and deduce it.